### ISSUE TITLE
Run tests for installed packages

### ISSUE THREAD

**Issue by Thom1729** [SublimeText/UnitTesting#155]
Currently, UnitTesting cannot run tests that are inside installed packages. This is because it inherits the `discover` method of `unittest.TestLoader`, which uses the filesystem directly.

We could reimplement `discover` in `UnitTestingLoader` to use the resource system instead. In principle, this should work for our use case, because every test suite that UnitTesting can find should be accessible as a resource.

This would allow package authors to verify that their tests run when the package is installed. End users could also more easily troubleshoot their installed packages.

I'm working on an implementation.

**Comment by randy3k** [comment]
Thank for the initiative work.

**Comment by Thom1729** [comment]
How would you feel about adding `sublime_lib` as a dependency? `sublime_lib.ResourcePath` would substantially simplify the implementation. (We could use other parts of it as well, but those would be separate PRs.)

**Comment by Thom1729** [comment]
I'm new to the CI stuff, but shouldn't Package Control install all of the dependencies automatically?

It looks like there's some pre-ST3 code in there. I wonder if it could be simplified a bit to take advantage of new features like the builtin "Install Package Control" command.

**Comment by randy3k** [comment]
Installation of Package Control is optional. If a user didn't install Package Control manually via `bootstrap install_package_control`, the script will download the current only dependency `coverage` from the github repo.

**Comment by Thom1729** [comment]
Good point. In that case, I would expect UnitTesting to behave as it normally would for a package with no tests.

Is there a good tutorial somewhere on setting up this sort of CI on a local machine for testing?

**Comment by randy3k** [comment]
I used Travis docker machine for testing.
